command/debug.go

```diff
@@ -524,140 +499,225 @@ func (c *DebugCommand) captureStaticTargets(client *api.Client) error {
+	intervalTicker <- time.Now()
+	tick := time.NewTicker(c.flagInterval)
+	go func() {
+		for t := range tick.C {
+			intervalTicker <- t
+		}
+	}()
 
-	if strutil.StrListContains(c.flagTargets, "host") {
-		c.logger.Info("capturing host information", "count", idxCount)
+	for {
+		select {
+		case <-ctx.Done():
+			tick.Stop()
+			return
+		case <-intervalTicker:
+			c.logger.Info("capturing host information", "count", idxCount)
+			idxCount++
 
-		wg.Add(1)
-		go func() {
-			r := client.NewRequest("GET", "/v1/sys/host-info")
-			resp, err := client.RawRequest(r)
+			r := c.client.NewRequest("GET", "/v1/sys/host-info")
+			resp, err := c.client.RawRequestWithContext(ctx, r)
 			if err != nil {
-				errCh <- newCaptureError("host", err)
+				c.captureError("host", err)
 			}
 			if resp != nil {
 				defer resp.Body.Close()
 
 				secret, err := api.ParseSecret(resp.Body)
 				if err != nil {
-					errCh <- newCaptureError("host", err)
+					c.captureError("host", err)
 				}
 				if hostEntry := secret.Data; hostEntry != nil {
-					c.hostInfoCh <- hostEntry
+					c.hostInfoCollection = append(c.hostInfoCollection, hostEntry)
 				}
 			}
-			wg.Done()
-		}()
+		}
 	}
+}
 
-	if strutil.StrListContains(c.flagTargets, "pprof") {
-		c.logger.Info("capturing pprof data", "count", idxCount)
+func (c *DebugCommand) collectMetrics(ctx context.Context) {
+	idxCount := 0
+
+	// intervalTicker needs to be buffered first in order to trigger a run
+	// right away.
+	intervalTicker := make(chan time.Time, 1)
+	intervalTicker <- time.Now()
+	tick := time.NewTicker(c.flagMetricsInterval)
+	go func() {
+		for t := range tick.C {
+			intervalTicker <- t
+		}
+	}()
+
+	for {
+		select {
+		case <-ctx.Done():
+			tick.Stop()
+			return
+		case <-intervalTicker:
+			c.logger.Info("capturing metrics", "count", idxCount)
+			idxCount++
+
+			healthStatus, err := c.client.Sys().Health()
+			if err != nil {
+				c.captureError("metrics", err)
+				continue
+			}
 
-		wg.Add(1)
-		go func() {
-			defer wg.Done()
+			// Check replication status. We skip on processing metrics if we're one
+			// of the following (since the request will be forwarded):
+			// 1. Any type of DR Node
+			// 2. Non-DR, non-performance standby nodes
+			switch {
+			case healthStatus.ReplicationDRMode == "secondary":
+				c.logger.Info("skipping metrics capture on DR secondary node")
+				continue
+			case healthStatus.Standby && !healthStatus.PerformanceStandby:
+				c.logger.Info("skipping metrics on standby node")
+				continue
+			}
+
+			// Perform metrics request
+			r := c.client.NewRequest("GET", "/v1/sys/metrics")
+			resp, err := c.client.RawRequestWithContext(ctx, r)
+			if err != nil {
+				c.captureError("metrics", err)
+				continue
+			}
+			if resp != nil {
+				defer resp.Body.Close()
+
+				metricsEntry := make(map[string]interface{})
+				err := json.NewDecoder(resp.Body).Decode(&metricsEntry)
+				if err != nil {
+					c.captureError("metrics", err)
```

Comment:
I would `continue` here if you got an error, rather than append a probably-nil metricsEntry.

Reply:
Yep, I’ll get this updated.